_rules/table-headers-attribute-refer-to-data-cells-a25f45.md

```diff
@@ -0,0 +1,287 @@
+- This test assumes that if the IDs of the `headers` attribute, do not live up to the expectations in this rule, the browser will try to calculate a default header from the [algorithm for assigning header cells](https://html.spec.whatwg.org/multipage/tables.html#header-and-data-cell-semantics).
+
+## Accessibility Support
+
+- `headers` attribute may not be consistently announced by assistive technologies.
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [F90: Incorrectly associating table headers and content via the headers and id attributes](https://www.w3.org/WAI/WCAG21/Techniques/failures/F90)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `headers` attribute on the cell refers to `th` within the same `table`.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th id="header1">Projects</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td headers="header1">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 2
+
+The `headers` attribute on the cell refers to `th` within the same `table`. Multiple headers are referenced for a cell with `colspan` of `2`.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th id="header1">Projects</th>
+      <th id="header2">Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2" headers="header1 header2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `headers` attribute on the cell refers to `td` with a role of `columnheader` within the same `table`.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <td role='columnheader' id="header1">Projects</td>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td headers="header1">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 4
+
+A `table` with multiple `columnheader(s)`, where the `headers` attribute on the cell lists the `id(s)` of `th` within the same `table`.
+
+```html
+<table>
+  <tr>
+    <th colspan="2" id="header1">Projects</th>
+    <th colspan="2" id="header2">Exams</th>
+  </tr>
+  <tr>
+    <th id="e1" headers="header1">1</th>
+    <th id="e2" headers="header1">2</th>
+    <th id="p1" headers="header2">1</th>
+    <th id="p2" headers="header2">2</th>
+  </tr>
+  <tr>
+    <td colspan="2" headers="header1 e1 e2">15%</td>
+    <td headers="header2 p1">15%</td>
+    <td headers="header2 p2">45%</td>
+  </tr>
+</table>
+```
+
+#### Passed Example 5
+
+The `headers` attribute on the cell refers to `th` with a role of `rowheader` within the same `table`.
+
+```html
+<table>
+  <tbody>
+    <tr>
+      <th role="rowheader" id="headerAge">Age</th>
+      <td headers="headerAge">65</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+### Failed
```

Comment:
Out of all of these, I think only failed example 3 actually fails SC 1.3.1. For the sake of having all failed examples be true failures of the SC, I think these things are going to need some work.